src/merkle/merkle-objects.ts

```diff
@@ -265,9 +268,14 @@ export class BloomMetadata implements MetadataFunction<Candidate, TreeMetadata>
         bloomFilterEntries.add(`controller-${controller}`)
       }
     }
-    const bloomFilter = BloomFilter.from(bloomFilterEntries, BLOOM_FILTER_FALSE_POSITIVE_RATE)
+
+    const bloomFilter = new BloomFilter(BLOOM_FILTER_FALSE_POSITIVE_RATE, bloomFilterEntries.size)
+    for (const entry of bloomFilterEntries) {
+      bloomFilter.add(entry)
+    }
+
     // @ts-ignore
-    const serializedBloomFilter = bloomFilter.saveAsJSON()
+    const serializedBloomFilter = bloomFilter.toString()
```

Comment:
Why is this a string and not a bytestring? Would be more efficient?

Reply:
I followed the example from https://crates.io/crates/growable-bloom-filter to serialize to JSON, but yes I suppose we can implement a custom serialization if needed.